From 62e687f06bec0edda85054517a9c26de9bd72d7c Mon Sep 17 00:00:00 2001
From: Niall Pemberton <niallp@apache.org>
Date: Wed, 2 Feb 2011 22:02:32 +0000
Subject: [PATCH] Port LANG-677 to 2.x branch - DateUtils isSameLocalTime()
 compares the hour using 12hour Calendar.HOUR instead of 24hour
 Calendar.HOUR_OF_DAY - thanks to Christian

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/lang/branches/LANG_2_X@1066653 13f79535-47bb-0310-9956-ffa450edef68
---
 src/main/java/org/apache/commons/lang/time/DateUtils.java | 2 +-
 .../java/org/apache/commons/lang/time/DateUtilsTest.java  | 8 ++++++++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/main/java/org/apache/commons/lang/time/DateUtils.java b/src/main/java/org/apache/commons/lang/time/DateUtils.java
index c62d93c4d6a..774735ab271 100644
--- a/src/main/java/org/apache/commons/lang/time/DateUtils.java
+++ b/src/main/java/org/apache/commons/lang/time/DateUtils.java
@@ -259,7 +259,7 @@ public static boolean isSameLocalTime(Calendar cal1, Calendar cal2) {
         return (cal1.get(Calendar.MILLISECOND) == cal2.get(Calendar.MILLISECOND) &&
                 cal1.get(Calendar.SECOND) == cal2.get(Calendar.SECOND) &&
                 cal1.get(Calendar.MINUTE) == cal2.get(Calendar.MINUTE) &&
-                cal1.get(Calendar.HOUR) == cal2.get(Calendar.HOUR) &&
+                cal1.get(Calendar.HOUR_OF_DAY) == cal2.get(Calendar.HOUR_OF_DAY) &&
                 cal1.get(Calendar.DAY_OF_YEAR) == cal2.get(Calendar.DAY_OF_YEAR) &&
                 cal1.get(Calendar.YEAR) == cal2.get(Calendar.YEAR) &&
                 cal1.get(Calendar.ERA) == cal2.get(Calendar.ERA) &&
diff --git a/src/test/java/org/apache/commons/lang/time/DateUtilsTest.java b/src/test/java/org/apache/commons/lang/time/DateUtilsTest.java
index 8eb17d8f131..51ca2c4a864 100644
--- a/src/test/java/org/apache/commons/lang/time/DateUtilsTest.java
+++ b/src/test/java/org/apache/commons/lang/time/DateUtilsTest.java
@@ -226,6 +226,14 @@ public void testIsSameLocalTime_Cal() {
         cal2.set(2004, 6, 9, 13, 45, 0);
         cal2.set(Calendar.MILLISECOND, 0);
         assertEquals(true, DateUtils.isSameLocalTime(cal1, cal2));
+
+        Calendar cal3 = Calendar.getInstance();
+        Calendar cal4 = Calendar.getInstance();
+        cal3.set(2004, 6, 9, 4,  0, 0);
+        cal4.set(2004, 6, 9, 16, 0, 0);
+        cal3.set(Calendar.MILLISECOND, 0);
+        cal4.set(Calendar.MILLISECOND, 0);
+        assertFalse("LANG-677", DateUtils.isSameLocalTime(cal3, cal4));
         
         cal2.set(2004, 6, 9, 11, 45, 0);
         assertEquals(false, DateUtils.isSameLocalTime(cal1, cal2));
